repos/config/detail/generate_repos_lock.bzl

```
@@ -0,0 +1,18 @@
+load("@python_deps//:requirements.bzl", "requirement")
+load("@rules_python//python:defs.bzl", "py_binary")
+
+def generate_repos_lock(*, name, repos_file, setup_file, overlay_files):
+    """Macro to create a py_binary for generating repos_lock.update"""
```

We don't need to put any implementation details into the macro documentation. They can read the source code if they want to know about that.
```suggestion
    """Executable rule for updating the `setup_file` from a `repos_file` and `overlay_files`."""
```